components/locale/src/subtags/variants.rs

```diff
@@ -26,7 +26,14 @@ use std::ops::Deref;
 ///
 /// [`Variant`]: ./struct.Variant.html
 #[derive(Default, Debug, PartialEq, Eq, Clone, Hash, PartialOrd, Ord)]
-pub struct Variants(Box<[Variant]>);
+pub struct Variants(
```

Comment:
Why didn't you make the `variants` field of `LanguageIdentifier` be `Option<Variants>` like the region and script subtags?  Why did you hide the `Option` down here?  Doing this would avoid the need to make any changes to this file.

Reply:
Mostly because logically there's no value in `Option` of collection since it doesn't have any separate meaning from an empty collection.

For that reason, my target is to have `variants` field be a collection, and in `Variants` I had to re-add the `Option` just because of the limitations of const function.
Since fields on `LanguageIdentifier` are public, having `Option` there makes it a public API, while having the `Option` here is an internal (except of from/to_raw) and I don't expect any customer to write logic which involves handling `variants` as an option of a collection.

Reply:
Ok.  Let's continue the discussion in #243